tasks/db_deploy/create_db.py

```diff
@@ -0,0 +1,144 @@
+        create_app_schema_role_users(conn, config)
+
+        # Change to DBO role and set search path
+        set_search_path_and_role(conn)
+
+        # Create the database objects
+        create_metadata_objects(conn)
+        create_recovery_objects(conn)
+
+        # If everything is good, commit.
+        conn.commit()
+
+
+def create_app_schema_role_users(connection, config):
+    """
+    Creates the ORCA application database schema, users and roles.
+
+    Args:
+        connection (sqlalchemy.engine.future.connect): Database connection.
+    """
+    # Create the roles first since they are needed by schema and users
+    logger.debug("Creating the ORCA dbo role ...")
+    connection.execute(dbo_role_sql())
+    logger.info("ORCA dbo role created.")
+
+    logger.debug("Creating the ORCA app role ...")
+    connection.execute(app_role_sql())
+    logger.info("ORCA app role created.")
+
+    # Create the schema next
+    logger.debug("Creating the ORCA schema ...")
+    connection.execute(orca_schema_sql())
+    logger.info("ORCA schema created.")
+
+    # Create the users last
+    logger.debug("Creating the ORCA application user ...")
+    connection.execute(app_user_sql(config["app_user_password"]))
+    logger.info("ORCA application user created.")
+
+
+def set_search_path_and_role(connection):
+    """
+    Sets the role to the dbo role to create/modify ORCA objects and sets the
+    search_path to make the orca schema first. This must be run before any
+    creations or modifications to ORCA objects in the ORCA schema.
+
+    Args:
+        connection (sqlalchemy.engine.future.connect): Database connection.
+    """
+    # Set the user and search_path for the install
+    logger.debug("Changing to the dbo role to create objects ...")
+    connection.execute(text("SET ROLE orca_dbo;"))
+
+    logger.debug("Setting search path to the ORCA schema to create objects ...")
+    connection.execute(text("SET search_path TO orca, public;"))
+
+
+def create_metadata_objects(connection):
+    """
+    Create the ORCA application metadata tables used to manage application
+    versions and other ORCA internal information in the proper order.
+    - schema_versions
+
+    Args:
+        connection (sqlalchemy.engine.future.connect): Database connection.
+    """
+    # Create metadata tables
+    logger.debug("Creating schema_versions table ...")
+    connection.execute(schema_versions_table_sql())
+    logger.info("schema_versions table created.")
+
+    # Populate the table with data
+    logger.debug("Populating the schema_versions table with data ...")
+    connection.execute(schema_versions_data_sql())
+    logger.info("Data added to the schema_versions table.")
+
+
+def create_recovery_objects(connection):
+    """
+    Creates the ORCA recovery tables in the proper order.
+    - recovery_status
+    - recovery_job
+    - recovery_table
+
+    Args:
+        connection (sqlalchemy.engine.future.connect): Database connection.
```

Comment:
Recommend using this as a type hint "connection: sqlalchemy.engine.future.connect" as it helps IDEs identify potential issues.

Reply:
Done